src/transformers/generation/utils.py

```diff
@@ -632,6 +632,8 @@ def _extract_past_from_model_output(self, outputs: ModelOutput, standardize_cach
             past_key_values = outputs.mems
         elif "past_buckets_states" in outputs:
             past_key_values = outputs.past_buckets_states
+        elif "cache_params" in outputs:
+            past_key_values = outputs.cache_params
```

Comment:
IMO we should not need to update the generation utils, unless we find cache params to be general enough?

Reply:
We need to pass in updated "cache" into every forward call for the model to actually "use_cache". So we have to update `model_kwargs` in generation utils. CRIIW, as I am assuming from the code that Mamba caching works same as Transformer models

As you said below, if we don't want to follow XLNet style, I can use its own naming for cache. I think in that case we have to make changes in XLNet and other models with different cache names, to make generation utils consistent. Otherwise we'll be using forcefully `past_key_values` for several models except Mamba, thus adding unnecessary conditional checks

Reply:
No, I think we ought to just leave previous model as is (they are using past_key_values) and just support arbitrary kwargs being returned from the model).